test/controllers/albums_controller_test.rb

```diff
@@ -0,0 +1,91 @@
+require 'test_helper'
+
+class AlbumsControllerTest < ActionController::TestCase
+  test "should get index" do
+    get :index
+    assert_response :success
+    assert_template :index
+  end
+
+  test "should get show" do
+    get :show, {id: albums(:suburbs).id}
+    assert_response :success
+    assert_template :show
+
+    assert_equal assigns(:album), albums(:suburbs)
+  end
+# not working
+  # test "should not get show for invalid id" do
+  #   get :show, {id: albums(:suburbs).id}
+  #   suburbs.delete
+  #   get :show, {id: albums(:suburbs).id}
+  #   assert_response :error
+  # end
+
+  test "should get edit" do
+    get :edit, {id: albums(:suburbs).id}
+    assert_response :success
+    assert_template :edit
+
+    assert_equal assigns(:album), albums(:suburbs)
+    assert_equal assigns(:album).recorded_by, "Arcade Fire"
+
+    assert_template partial: '_albumform', count: 1
+  end
+
+  test "should be able to update" do
+    patch :update, :id => albums(:suburbs), :album => {:title => 'Suburbs'}
+    assert_equal "Suburbs", assigns(:album).title
```

Comment:
This is an excellent test to do!

For more flexibility I would recommend changing `assigns(:album).title` to `albums(:suburbs).reload.title`.

When we use `assigns(:album)` we're requiring that the controller sets an instance variable called `@album`. While this is certainly the case with your current code, if that variable name changes but the behavior of the application doesn't, our test will still fail.

Instead, if we use `albums(:suburbs).reload` we are using the fixture definition and simply forcing it to reload the most recent data about that album from the database. This is more flexible and doesn't depend upon the details of the controller's implementation.